plugins/inputs/icinga2/icinga2.go

```diff
@@ -123,39 +209,70 @@ func (i *Icinga2) Gather(acc telegraf.Accumulator) error {
 		i.client = client
 	}
 
-	requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
-
-	// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
-	//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
-	if i.ObjectType == "services" {
-		requestURL += "&attrs=host_name"
+	// For backward config compatibility
+	if i.ObjectType != "" {
+		i.Objects = []string{i.ObjectType}
 	}
 
-	address := fmt.Sprintf(requestURL, i.Server, i.ObjectType)
+	// Collect /v1/objects
+	for _, objectType := range i.Objects {
+		requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
 
-	req, err := http.NewRequest("GET", address, nil)
-	if err != nil {
-		return err
-	}
+		// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
+		//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
+		if objectType == "services" {
+			requestURL += "&attrs=host_name"
+		}
 
-	if i.Username != "" {
-		req.SetBasicAuth(i.Username, i.Password)
-	}
+		address := fmt.Sprintf(requestURL, i.Server, objectType)
 
-	resp, err := i.client.Do(req)
-	if err != nil {
-		return err
-	}
+		resp, err := i.icingaRequest(address)
+		if err != nil {
+			return err
+		}
 
-	defer resp.Body.Close()
+		result := ResultObject{}
+		err = i.parseObjectResponse(resp, &result)
+		if err != nil {
+			return err
+		}
 
-	result := Result{}
-	err = json.NewDecoder(resp.Body).Decode(&result)
-	if err != nil {
-		return err
+		i.gatherObjects(acc, result, objectType)
 	}
 
-	i.GatherStatus(acc, result.Results)
+	// Collect /v1/status
+	for _, statusType := range i.Status {
+		address := fmt.Sprintf("%s/v1/status/%s", i.Server, statusType)
+
+		resp, err := i.icingaRequest(address)
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"component": statusType,
+		}
+		var fields map[string]interface{}
+
+		switch statusType {
+		case "ApiListener":
+			fields, err = i.parsePerfdataResponse(resp)
+		case "CIB":
+			fields, err = i.parseCIBResponse(resp)
+		case "IdoMysqlConnection":
+			fields, err = i.parsePerfdataResponse(resp)
+		case "IdoPgsqlConnection":
+			fields, err = i.parsePerfdataResponse(resp)
+		default:
+			return fmt.Errorf("status endpoint %s is not supported", statusType)
+		}
```

Comment:
This is also something to be checked during Init. For example with the internal `choice.CheckSlice()`.

Reply:
Ok, thanks for the hint!